clients/src/main/java/org/apache/kafka/common/requests/ListOffsetsRequest.java

```diff
@@ -65,17 +66,25 @@ public static Builder forReplica(short allowedVersion, int replicaId) {
         public static Builder forConsumer(boolean requireTimestamp,
```

Comment:
There are too many boolean arguments. Could you please refactor it? for instance:
```java
        public static Builder forTieredStorageTimestamp(IsolationLevel isolationLevel) {

            return new Builder((short) 9, ApiKeys.LIST_OFFSETS.latestVersion(), CONSUMER_REPLICA_ID, isolationLevel);
        }
```